# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieNotification.h
# Commit: 6379a4984eba
# Full Hash: 6379a4984ebafb659428ec3e4493f72173b3a70a
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-22 04:22:03
# Description:
#   Bug 1854028 - Update BounceTrackingStorageObserver cookie check and remove nsICookieNotification isThirdPartyCookie. r=bvandersloot,cookie-reviewers,anti-tracking-reviewers,timhuang
#   
#   For the BounceTrackingStorageObserver, instead of passing in a third-party flag
#   we can simply check if the changed cookie is partitioned or not.
#   
# ==============================================================================

diff -r 24c5f04d3793 -r 6379a4984eba netwerk/cookie/CookieNotification.h
--- a/netwerk/cookie/CookieNotification.h	Thu Sep 21 14:30:17 2023 +0000
+++ b/netwerk/cookie/CookieNotification.h	Thu Sep 21 14:42:32 2023 +0000
@@ -24,14 +24,12 @@
   explicit CookieNotification(nsICookieNotification::Action aAction,
                               nsICookie* aCookie, const nsACString& aBaseDomain,
                               nsIArray* aBatchDeletedCookies = nullptr,
-                              uint64_t aBrowsingContextId = 0,
-                              bool aIsThirdPartyCookie = false)
+                              uint64_t aBrowsingContextId = 0)
       : mAction(aAction),
         mCookie(aCookie),
         mBaseDomain(aBaseDomain),
         mBatchDeletedCookies(aBatchDeletedCookies),
-        mBrowsingContextId(aBrowsingContextId),
-        mIsThirdPartyCookie(aIsThirdPartyCookie){};
+        mBrowsingContextId(aBrowsingContextId){};
 
  private:
   nsICookieNotification::Action mAction;
@@ -39,7 +37,6 @@
   nsCString mBaseDomain;
   nsCOMPtr<nsIArray> mBatchDeletedCookies;
   uint64_t mBrowsingContextId = 0;
-  bool mIsThirdPartyCookie = false;
 
   ~CookieNotification() = default;
 };